editpost.php

```diff
@@ -580,8 +580,7 @@
 	eval("\$loginbox = \"".$templates->get("changeuserbox")."\";");
 
 	$deletebox = '';
-	// Can we delete posts?
-	if($post['visible'] != -1 && (is_moderator($fid, "candeleteposts") || $forumpermissions['candeleteposts'] == 1 && $mybb->user['uid'] == $post['uid']))
+	if(($thread['firstpost'] == $pid && $post['visible'] != -1 && (is_moderator($fid, "candeletethreads")) || ($forumpermissions['candeletethreads'] == 1 && $mybb->user['uid'] == $post['uid'] || $post['visible'] != -1 && (is_moderator($fid, "candeleteposts") || $forumpermissions['candeleteposts'] == 1 && $mybb->user['uid'] == $post['uid'])) 
```

Comment:
Looks like it should be

```suggestion
	if($post['visible'] != -1 && (($thread['firstpost'] == $pid && (is_moderator($fid, "candeletethreads") || $forumpermissions['candeletethreads'] == 1 && $mybb->user['uid'] == $post['uid'])) || (is_moderator($fid, "candeleteposts") || $forumpermissions['candeleteposts'] == 1 && $mybb->user['uid'] == $post['uid'])))
```

Reply:
Right that makes sense when looking at https://github.com/mybb/mybb/blob/76bd4568a4cbbf35b6b3922b8605e40809d7d6d2/editpost.php#L143

But I think it should be the following because the last part should only pass if the post isn't the first in the thread.

<details>
<summary>

```suggestion
	if($post['visible'] != -1 && (($thread['firstpost'] == $pid && (is_moderator($fid, "candeletethreads") || $forumpermissions['candeletethreads'] == 1 && $mybb->user['uid'] == $post['uid'])) || ($thread['firstpost'] != $pid && (is_moderator($fid, "candeleteposts") || $forumpermissions['candeleteposts'] == 1 && $mybb->user['uid'] == $post['uid'])))) 
```
</summary>

```
(
    $post['visible'] != -1 &&
    (
        (
            $thread['firstpost'] == $pid &&
            (
                is_moderator($fid, "candeletethreads") || $forumpermissions['candeletethreads'] == 1 && $mybb->user['uid'] == $post['uid']
            )
        ) || (
            $thread['firstpost'] != $pid &&
            (
                is_moderator($fid, "candeleteposts") || $forumpermissions['candeleteposts'] == 1 && $mybb->user['uid'] == $post['uid']
            )
        )
    )
)
```
</details>